crates/starknet-server/src/api/json_rpc/models.rs

```diff
@@ -457,9 +452,9 @@ mod tests {
             r#"{"block_id": {"block_hash": "0x004134134134134134134134134134134134134134134134134134134134134134"}}"#,
         );
 
-        // Block hash hex doesnt start with 0x
+        // Changing to starknet_rs BlockId, now those cases can be handled - hex to be non prefixed
         assert_block_id_block_hash_correctness(
-            false,
+            true,
             "0x01",
             r#"{"block_id": {"block_hash": "01"}}"#,
```

Comment:
But should this really be correct? What is this test testing? The correctness of `FieldElement` implementation or something in our code?

Reply:
In our implementation hex strings which were not prefixed with 0x were not possible. Starknet-rs accept such hex values

Reply:
Ok, are we then diverging from the specification in some cases? By not being strict enough?